### PR TITLE
Add patterns for fluentd images

### PR DESCRIPTION
**Pull request by omegas27**
_Retagger_

https://github.com/giantswarm/giantswarm/blob/master/ops-recipes/container-registry.md

---

### CODE REVIEW

**Comment by axbarsan** [comment]
If I understand correctly, this will automatically update the fluentd images in our quay repo, when there's a new one released, right?

**Comment by omegas27** [comment]
As long as they match the pattern, yes :) but it will also upload them to alyun in China

File: images.yaml [hunk 1]
```diff
@@ -68,17 +68,15 @@
...
 - name: fluent/fluent-bit-0.13-dev
   tags:
   - sha: 0ae482ad3e8d951a66090b968389f2a11b0b16b8a208b5edb4ccb0ca5800b90d
     tag: "0.18"
 - name: fluent/fluentd-kubernetes-daemonset
-  tags:
-  - sha: f9fa60dab5e7a6ae4adbe972a2fc22cf7eaa148df74cde4cd7f57e7019260451
-    tag: "v1.3-debian-cloudwatch"
+  patterns:
+  - pattern: '>= 1.[0-9]+.[0-9]+-debian-cloudwatch-1.0'
```
Comment (MarcelMue) [inline images.yaml:79]: Careful with this one - we use a semver comparison and this might try to tag all pre release versions. Let me check.

Comment (omegas27) [reply images.yaml:79]: Maybe I can adjust the pattern to 1.[3-9][0-9]*.[0-9]+-debian-cloudwatch-1.0 ?

Comment (MarcelMue) [reply images.yaml:79]:
```
{"caller":"retagger/main.go:50","level":"error","message":"failed to execute command","stack":"[{/home/marcel/golang/src/github.com/giantswarm/retagger/cmd/runner.go:36: } {/home/marcel/golang/src/github.com/giantswarm/retagger/cmd/runner.go:98: } {/home/marcel/golang/src/github.com/giantswarm/retagger/pkg/retagger/executor.go:17: } {/home/marcel/golang/src/github.com/giantswarm/retagger/pkg/retagger/job_pattern.go:46: } {/home/marcel/golang/src/github.com/giantswarm/retagger/pkg/retagger/job_pattern.go:105: } {improper constraint: \u003e= 1.[0-9]+.[0-9]+}]","time":"2020-04-08T08:06:38.962915+00:00"}
```
I don't think semver likes this :(

**Comment by MarcelMue** [comment]
See here:
https://github.com/Masterminds/semver#basic-comparisons

File: images.yaml (hunk 1 above)
Comment (omegas27) [reply images.yaml:79]: Ok, so it is not a regex pattern. I guess 1.x.x would be better then

**Comment by omegas27** [comment]
@stone-z I need it for an app. I will check batman or halo has some automation to update apps automatically

File: images.yaml [hunk 2]
```diff
@@ -68,17 +68,15 @@
...
 - name: fluent/fluent-bit-0.13-dev
   tags:
   - sha: 0ae482ad3e8d951a66090b968389f2a11b0b16b8a208b5edb4ccb0ca5800b90d
     tag: "0.18"
 - name: fluent/fluentd-kubernetes-daemonset
-  tags:
-  - sha: f9fa60dab5e7a6ae4adbe972a2fc22cf7eaa148df74cde4cd7f57e7019260451
-    tag: "v1.3-debian-cloudwatch"
+  patterns:
+  - pattern: '>= 1.x.x-debian-cloudwatch-1.0'
```
Comment (MarcelMue) [inline images.yaml:79]: First one now works, second one it now tags all `pre-release` tags :
```
{"caller":"github.com/giantswarm/retagger/pkg/retagger/job_pattern.go:52","level":"debug","message":"Found 550 upstream tags which match the pattern \u003e= 1.x.x-debian-cloudwatch-1.0","time":"2020-04-08T08:39:49.395126+00:00"}
```
So maybe keep the first pattern and set them explicitly here until we come up with a solution for pre-releases.

Comment (omegas27) [reply images.yaml]: Thanks :)

**Pull request merged**